test/mysql-cdc/alter-table-after-source.td

```diff
@@ -0,0 +1,413 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ set-sql-timeout duration=1s
+$ set-max-tries max-tries=20
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+alter system set enable_mysql_source = true
+
+#
+# Test ALTER TABLE -- source will error out for tables which existed when the source was created
+#
+
+> CREATE SECRET mysqlpass AS '${arg.mysql-root-password}'
+> CREATE CONNECTION mysql_conn TO MYSQL (
+    HOST mysql,
+    USER root,
+    PASSWORD SECRET mysqlpass
+  )
+
+$ mysql-connect name=mysql url=mysql://root@mysql password=${arg.mysql-root-password}
+
+$ mysql-execute name=mysql
+DROP DATABASE IF EXISTS public;
+CREATE DATABASE public;
+USE public;
+
+CREATE TABLE add_columns (f1 INTEGER);
+INSERT INTO add_columns VALUES (1);
+
+CREATE TABLE remove_column (f1 INTEGER, f2 VARCHAR(2));
+INSERT INTO remove_column VALUES (2, 'ab');
+
+CREATE TABLE alter_column (f1 INTEGER, f2 VARCHAR(2));
+INSERT INTO alter_column VALUES (2, 'ab');
+
+CREATE TABLE alter_drop_nullability (f1 INTEGER NOT NULL);
+INSERT INTO alter_drop_nullability VALUES (1);
+
+CREATE TABLE alter_add_nullability (f1 INTEGER);
+INSERT INTO alter_add_nullability VALUES (1);
+
+CREATE TABLE alter_drop_pk (f1 INTEGER PRIMARY KEY);
+INSERT INTO alter_drop_pk VALUES (1);
+
+CREATE TABLE alter_add_pk (f1 INTEGER);
+INSERT INTO alter_add_pk VALUES (1);
+
+CREATE TABLE alter_cycle_pk (f1 INTEGER PRIMARY KEY);
+INSERT INTO alter_cycle_pk VALUES (1);
+
+CREATE TABLE alter_cycle_pk_off (f1 INTEGER);
+INSERT INTO alter_cycle_pk_off VALUES (1);
+
+CREATE TABLE alter_drop_unique (f1 INTEGER UNIQUE);
+INSERT INTO alter_drop_unique VALUES (1);
+
+CREATE TABLE alter_add_unique (f1 INTEGER);
+INSERT INTO alter_add_unique VALUES (1);
+
+CREATE TABLE alter_extend_column (f1 VARCHAR(2));
+INSERT INTO alter_extend_column VALUES ('ab');
+
+# TODO: #25032 (decimal type)
```

Comment:
this should be possible on latest main

Reply:
Thanks! (By the way, the closed-issue-detector job in nightly would have complained when the referenced issue is closed 😀.)

✅ Addressed.
I will add further test for decimal as follow-up.